src/__internal__/popover/popover.component.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect, useRef } from "react";
+import ReactDOM from "react-dom";
+import PropTypes from "prop-types";
+import { createPopper } from "@popperjs/core";
+
+const Popover = ({
+  children,
+  placement,
+  options,
+  disablePortal,
+  reference,
+}) => {
+  const elementDOM = useRef();
+  if (!elementDOM.current) {
+    elementDOM.current = document.createElement("div");
+  }
+  const popperInstance = useRef();
+  const popperRef = useRef();
+  const popperElement = React.Children.only(children).ref || popperRef;
+
+  useEffect(() => {
+    popperInstance.current = createPopper(
+      reference.current,
+      popperElement.current,
+      {
+        placement,
+        ...options,
+      }
+    );
+
+    return () => {
+      popperInstance.current.destroy();
+      popperInstance.current = null;
+    };
+  }, [options, placement, popperElement, reference]);
+
+  // eslint-disable-next-line consistent-return
+  useEffect(() => {
+    if (!disablePortal) {
+      const portalElement = elementDOM.current;
+      document.body.appendChild(portalElement);
+      return () => {
+        document.body.removeChild(portalElement);
+      };
+    }
+  }, [disablePortal]);
+
+  const content = React.cloneElement(children, { ref: popperElement });
```

Comment:
could you simplify that to be clearer what is happening there? I think that could be done by an assertion, what do you think?

```
const popperRef = useRef();
let content;
let popperElementRef;

if (React.Children.only(children).ref) {
  content = children;
  popperElementRef = React.Children.only(children).ref;
} else {
  content = React.cloneElement(children, { ref: popperRef });
  popperElementRef = popperRef;
}
```